File: src/main/java/br/com/blu/bs2/sintetizador/contas/csv/service/strategy/BuilderBonusContaNaoEspecificada.java
package br.com.blu.bs2.sintetizador.contas.csv.service.strategy;

import br.com.blu.bs2.sintetizador.contas.csv.model.dominio.Conta;

import java.math.BigDecimal;

public class BuilderBonusContaNaoEspecificada implements IFactoryObterPercentualBonusService {

    private static final double VALOR_FIXO_DEFAUL_PERCENTUAL_ZERADO = 0.0;

    @Override
    public boolean isAppliable(Conta conta) {
        return false;
    }

    @Override
    public BigDecimal obterValorPercentualBonusAoDeposito(Conta conta) {
        return BigDecimal.valueOf(VALOR_FIXO_DEFAUL_PERCENTUAL_ZERADO);
    }

    @Override
    public PercentualBonus getPercentualBonus(Conta conta) {
        return PercentualBonus.BONUS_CONTA_NAO_ESPECIFICADA;
    }
}

File: src/main/kotlin/br/com/blu/bs2/sintetizador/contas/csv/service/strategy/BuilderBonusContaNaoEspecificada.kt
package br.com.blu.bs2.sintetizador.contas.csv.service.strategy

import br.com.blu.bs2.sintetizador.contas.csv.model.dominio.Conta
import br.com.blu.bs2.sintetizador.contas.csv.service.strategy.PercentualBonus
import java.math.BigDecimal

class BuilderBonusContaNaoEspecificada : IFactoryObterPercentualBonusService {
    override fun isAppliable(conta: Conta?): Boolean {
        return false
    }

    override fun obterValorPercentualBonusAoDeposito(conta: Conta?): BigDecimal? {
        return BigDecimal.valueOf(VALOR_FIXO_DEFAUL_PERCENTUAL_ZERADO)
    }

    override fun getPercentualBonus(conta: Conta?): PercentualBonus? {
        return PercentualBonus.BONUS_CONTA_NAO_ESPECIFICADA
    }

    companion object {
        private const val VALOR_FIXO_DEFAUL_PERCENTUAL_ZERADO = 0.0
    }
}
